lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -1166,6 +1166,9 @@ bool swift::diagnoseNonSendableTypesInReference(
     if (funcCheckOptions.contains(FunctionCheckKind::Params)) {
       // only check params if funcCheckKind specifies so
       for (auto param : *function->getParameters()) {
+        if (param->isSending())
+          return true;
```

Comment:
Here you are skipping sendability validation on all other parameters if at least one of them is marked as `sending`, which seems incorrect, we should check sendability of all arguments.
Also this function is supposed to return `true` if there are any errors being produced, which is not what happens in our case.
It seems like instead of instead of bailing out early from this function we should keep checking other arguments but skip `diagnoseNonSendableTypes` check below if the argument is marked as `sending`.

Could you also consider adding a test for a case where first argument is marked with `sending` but a second argument isn't?

Reply:
Thank you for the review.

> we should check sendability of all arguments.

I think that's right. Let me check it again.

Also, I might be misunderstanding this comment, but are you talking about `type->hasError()` in `diagnoseSingleNonSendableType`? If so, I think I overlooked it and we need to check `type->hasError()` and return `false` if there is any error.

```
static bool diagnoseSingleNonSendableType(
    Type type, SendableCheckContext fromContext,
    Type inDerivedConformance, SourceLoc loc,
    llvm::function_ref<bool(Type, DiagnosticBehavior)> diagnose) {
  if (type->hasError())
    return false;
```

Reply:
I fixed it in https://github.com/swiftlang/swift/pull/78601/commits/f63350561e7f04bfc9fdcd8497d8dae124749051. Please check it when you have time. Thank you!